rasa/server.py

```diff
@@ -466,7 +466,30 @@ def create_app(
             "to the state of a conversation.",
         )
 
+        verbosity = event_verbosity_parameter(request, EventVerbosity.AFTER_RESTART)
+
+        try:
+            async with app.agent.lock_store.lock(conversation_id):
+                tracker = await get_tracker(
+                    app.agent.create_processor(), conversation_id
+                )
+
+                events = _get_events_from_request_body(request)
```

Comment:
this can / should happen before the tracker is fetched, for instance right away in line 468

Reply:
I added a comment to clarify why I'm doing it like this. 
The problem is that the events get assigned new timestamps when they are parsed from the request payload. However, also the tracker initialization might lead to events such as the session started event which should be **before** the appended events. Does that make sense?

Reply:
makes sense, thanks for adding that comment